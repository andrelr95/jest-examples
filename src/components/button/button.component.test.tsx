import React from 'react'
import { act, fireEvent, render, screen, cleanup } from '@testing-library/react'
import { create } from 'react-test-renderer'

import Button from './button.component'

afterEach(cleanup)

describe('<Button /> tests', () => {
  it('should exists', () => {
    expect(Button).toBeDefined()
  })
  it('should trigger onClick prop', () => {
    // Given
    const onClickMock = jest.fn()
    render(<Button onClick={onClickMock}>Button name</Button>)

    // When
    act(() => {
      fireEvent.click(screen.getByRole('button', { name: /button name/i }))
    })

    // Then
    expect(onClickMock).toHaveBeenCalled()
  })

  it('should render button color variations', () => {
    // Given
    render(
      <Button
        onClick={() => {}}
        backgroundColor="blue"
        color="white"
      >
        Button name
      </Button>
    )

    // When
    const button = screen.getByRole('button', { name: /button name/i })

    // Then
    expect(button).toHaveStyle('background-color: blue')
    expect(button).toHaveStyle('color: white')
  })

  it('should render correctly the snapshot', () => {
    const component = create(
      <Button 
        color="white" 
        backgroundColor="red"
        onClick={() => {}}
      >
        Button name
      </Button>
    )

    component.toJSON()

    expect(component).toMatchSnapshot()
  })
})