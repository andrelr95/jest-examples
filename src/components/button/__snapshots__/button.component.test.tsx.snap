// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> tests should render correctly the snapshot 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "red",
      "color": "white",
    }
  }
>
  Button name
</button>
`;
